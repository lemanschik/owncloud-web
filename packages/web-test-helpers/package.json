{
  "name": "web-test-helpers",
  "private": true,
  "main": "src/index.ts",
  "peerDependencies": {
    "@casl/ability": "^6.3.3",
    "@casl/vue": "^2.2.1",
    "axios": "0.27.2",
    "design-system": "@ownclouders/design-system",
    "vue3-gettext": "^2.3.3",
    "vue-router": "4.1.6",
    "vuex": "4.1.0"
  }
}
