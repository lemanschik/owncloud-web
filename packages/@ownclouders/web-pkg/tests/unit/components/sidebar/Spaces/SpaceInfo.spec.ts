import SpaceInfo from 'web-pkg/src/components/sideBar/Spaces/SpaceInfo.vue'
import { defaultPlugins, shallowMount } from 'web-test-helpers'

const spaceMock = {
  type: 'space',
  name: ' space',
  id: '1',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  spaceQuota: {
    used: 100
  }
}

const selectors = {
  name: '[data-testid="space-info-name"]',
  subtitle: '[data-testid="space-info-subtitle"]'
}

describe('SpaceInfo', () => {
  it('shows space info', () => {
    const { wrapper } = createWrapper(spaceMock)
    expect(wrapper.find(selectors.name).exists()).toBeTruthy()
    expect(wrapper.find(selectors.subtitle).exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})

function createWrapper(spaceResource) {
  return {
    wrapper: shallowMount(SpaceInfo, {
      global: {
        plugins: [...defaultPlugins()],
        provide: { resource: spaceResource }
      }
    })
  }
}
