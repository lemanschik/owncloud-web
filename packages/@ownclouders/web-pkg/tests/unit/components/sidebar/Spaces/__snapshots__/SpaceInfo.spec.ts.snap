// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceInfo shows space info 1`] = `
<div class="space_info oc-p-s">
  <div class="space_info__body oc-text-overflow oc-flex oc-flex-middle">
    <div class="oc-mr-s">
      <oc-icon-stub accessiblelabel="" class="oc-display-block" color="" filltype="fill" name="layout-grid" size="medium" type="span" variation="passive"></oc-icon-stub>
    </div>
    <div>
      <h3 class="oc-font-semibold oc-m-rm"> space</h3>
      <span></span>
    </div>
  </div>
</div>
`;
